property.go

```diff
@@ -326,45 +392,118 @@ func parseProperties(raw map[string]interface{}) (map[string]Property, error) {
 					p = &RichTextProperty{}
 				}
 			case PropertyTypeSelect:
-				selectMap, ok := v.(map[string]interface{})["select"].(map[string]interface{})
-				if !ok {
-					return nil, errors.Errorf("an error occured while parsing property type: %s", rawProperty)
-				}
-				_, found := selectMap["options"]
-				if found {
-					p = &SelectProperty{}
-				} else {
-					p = &SelectOptionProperty{}
-				}
+				p = &DatabaseSelectProperty{}
 			case PropertyTypeMultiSelect:
-				switch v.(map[string]interface{})["multi_select"].(type) {
+				p = &DatabaseMultiSelectProperty{}
+			case PropertyTypeNumber:
+				p = &NumberProperty{}
+			case PropertyTypeCheckbox:
+				p = &DatabaseCheckboxProperty{}
+			case PropertyTypeEmail:
+				p = &DatabaseEmailProperty{}
+			case PropertyTypeURL:
+				p = &DatabaseURLProperty{}
+			case PropertyTypeFile:
+				p = &DatabaseFilesProperty{}
+			case PropertyTypePhoneNumber:
+				p = DatabasePhoneNumberProperty{}
+			case PropertyTypeFormula:
+				p = &FormulaProperty{}
+			case PropertyTypeDate:
+				p = &DatabaseDateProperty{}
+			case PropertyTypeRelation:
+				p = &RelationProperty{}
+			case PropertyTypeRollup:
+				p = &RollupProperty{}
+			case PropertyTypePeople:
+				p = &DatabasePeopleProperty{}
+			case PropertyTypeCreatedTime:
+				p = &CreatedTimeProperty{}
+			case PropertyTypeCreatedBy:
+				p = &CreatedByProperty{}
+			case PropertyTypeLastEditedTime:
+				p = &LastEditedTimeProperty{}
+			case PropertyTypeLastEditedBy:
+				p = &LastEditedByProperty{}
+			default:
+				return nil, errors.New(fmt.Sprintf("unsupported property type: %s", rawProperty["type"].(string)))
+			}
+			b, err := json.Marshal(rawProperty)
+			if err != nil {
+				return nil, err
+			}
+
+			if err = json.Unmarshal(b, &p); err != nil {
+				return nil, err
+			}
+
+			result[k] = p
+		default:
+			return nil, errors.New(fmt.Sprintf("unsupported property format %T", v))
+		}
+	}
+
+	return result, nil
+}
+
+type PageProperties map[string]Property
```

Comment:
`PageProperties` being a map of `Property` means a user can assign any `DatabaseProperty` in there, leading to invalid requests, that will be rejected. 

What about having two interfaces for those two kind of properties instead of one?